pkg/jsonnet/find_importers.go

```diff
@@ -220,8 +220,24 @@ func findImporters(root string, searchForFile string, chain map[string]struct{})
 		}
 	}
 
-	importersCache[key] = importers
-	return importers, nil
+	// If we've found a vendored file, check that it's not overridden by a vendored file in the environment root
+	// In that case, we only want to keep the environment vendored file
+	var filteredImporters []string
+	rootVendor := filepath.Join(root, "vendor")
+	if strings.HasPrefix(searchForFile, rootVendor) {
+		for _, importer := range importers {
+			relativePath := strings.TrimPrefix(searchForFile, rootVendor)
+			relativePath = strings.TrimPrefix(relativePath, "/")
+			if _, ok := jsonnetFilesCache[root][filepath.Join(filepath.Dir(importer), "vendor", relativePath)]; !ok {
```

Comment:
nit: probably be clearer (to me at least) if you named the local vendored file so this became something like

```suggestion
			if _, ok := jsonnetFilesCache[root][vendoredFileInEnvironment]; !ok {
```

or similar